Azure/azure-cli
# [AppService] Fix #14788: az webapp create not getting correct webapp when names are substrings

**Description**  
Fixes #14788 
When there are multiple webapps with names that are substrings of each other, in certain cases `az webapp create` uses the wrong webapp. 

**Testing Guide**  
- Create different webapps with names that are substrings of each other, for ex. "test-webap", "test-weba", "test-webapp" in different resource groups
`az webapp create -g RG1 --plan Plan1 --name test-webap`
`az webapp create -g RG2 --plan Plan2 --name test-weba`
`az webapp create -g RG3 --plan Plan3 --name test-webapp`

- Create the second webapp again. The command should find the correct existing webapp and update the correct one.
`az webapp create -g RG2 --plan Plan2 --name test-weba`

Expected behaviour: Re running the create commands should find the correct existing webapps and update them.
Current behaviour: Re running the create commands uses the wrong webapp from different resource group, and throws error:

`The webapp 'test-webap' exists in resource group 'RG1' and does not match the value entered 'RG2'. Please re-run command with the correct parameters.`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
